### ISSUE TITLE
"E565: Not allowed to change text"

### ISSUE THREAD

**Issue by justinmk** [lewis6991/satellite.nvim#76]
**Describe the bug**

Observed when using fzf to open a file (seems that gitsigns is also involved):

```
Error detected while processing function 30[30]..<SNR>44_callback[25]..function 30[30]..<SNR>44_callback:                                                                                                            
line   23:                                                                                                                                                                                                           
Vim(append):Error executing lua callback: ...te/pack/paqs/start/satellite.nvim/lua/satellite/view.lua:37: E565: Not allowed to change text or change window                                                          
stack traceback:                                                                                                                                                                                                     
        [C]: in function 'nvim_open_win'                                                                                                                                                                             
        ...te/pack/paqs/start/satellite.nvim/lua/satellite/view.lua:37: in function 'f'                                                                                                                              
        ...te/pack/paqs/start/satellite.nvim/lua/satellite/util.lua:261: in function 'create_view'                                                                                                                   
        ...te/pack/paqs/start/satellite.nvim/lua/satellite/view.lua:119: in function 'get_or_create_view'                                                                                                            
        ...te/pack/paqs/start/satellite.nvim/lua/satellite/view.lua:248: in function <...te/pack/paqs/start/satellite.nvim/lua/satellite/view.lua:242>                                                               
        [C]: in function 'nvim_exec_autocmds'                                                                                                                                                                        
        ...te/pack/paqs/start/gitsigns.nvim/lua/gitsigns/status.lua:15: in function 'autocmd_update'                                                                                                                 
        ...te/pack/paqs/start/gitsigns.nvim/lua/gitsigns/status.lua:49: in function 'clear'                                                                                                                          
        ...te/pack/paqs/start/gitsigns.nvim/lua/gitsigns/attach.lua:403: in function 'detach'                                                                                                                        
        ...te/pack/paqs/start/gitsigns.nvim/lua/gitsigns/attach.lua:121: in function <...te/pack/paqs/start/gitsigns.nvim/lua/gitsigns/attach.lua:119>
```

**To Reproduce**

I don't have consistent repro steps.

**Expected behavior**

Generally if satellite has the wrong window or buffer, it should just skip its update, perhaps logging a warning.

**Comment by justinmk** [comment]
I've also noticed that when a satellite error happens it seems like 'eventignore' is not restored. So at least this code should use pcall() to ensure that eventignore is restored? https://github.com/lewis6991/satellite.nvim/blob/dfc2b220f4969ebc49b0f2a94ab6faae240705bd/lua/satellite/util.lua#L260-L262

**Comment by lewis6991** [comment]
Attempted to fix in #77
